tox.ini

```diff
@@ -1,68 +1,41 @@
 [flake8]
-max-line-length=170
 exclude=connexion/__init__.py
-rst-roles=class
+rst-roles=class,mod,obj
+# https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#flake8
+# Longest docstring in current code base
+max-line-length=137
+extend-ignore=E203,RST303
 
 [tox]
+isolated_build = True
 envlist =
-    {py37}-{min,pypi,dev}
-    {py38}-{min,pypi,dev}
-    {py39}-{min,pypi,dev}
-    isort-check
-    isort-check-examples
-    isort-check-tests
-    flake8
-    mypy
+    {py39,py310,py311,py312}-pypi
+    pre-commit
 
 [gh-actions]
 python =
-    3.7: py37-min,py37-pypi
-    3.8: py38-min,py38-pypi
-    3.9: py39-min,py39-pypi,flake8,isort-check,isort-check-examples,isort-check-tests,mypy
+    3.9: py39-pypi
+    3.10: py310-pypi
+    3.11: py311-pypi,pre-commit
+    3.12: py312-pypi
 
 [testenv]
 setenv=PYTHONPATH = {toxinidir}:{toxinidir}
-deps=pytest
-commands=
-    pip install Requirements-Builder
-    min: requirements-builder --level=min --extras aiohttp -o {toxworkdir}/requirements-min.txt setup.py
-    min: pip install --upgrade -r {toxworkdir}/requirements-min.txt
-    pypi: requirements-builder --level=pypi --extras aiohttp -o {toxworkdir}/requirements-pypi.txt setup.py
-    pypi: pip install --upgrade -r {toxworkdir}/requirements-pypi.txt
-    dev: requirements-builder --level=dev --extras aiohttp --req=requirements-devel.txt -o {toxworkdir}/requirements-dev.txt setup.py
-    dev: pip install --upgrade -r {toxworkdir}/requirements-dev.txt
-    python setup.py test
-
-[testenv:flake8]
-deps=
-    flake8==3.9.2
-    flake8-rst-docstrings==0.2.3
-commands=python setup.py flake8
-
-[testenv:isort-check]
-basepython=python3
-deps=isort==5.9.1
-changedir={toxinidir}/connexion
-commands=isort --project connexion --check-only --diff .
-
-[testenv:isort-check-examples]
-basepython=python3
-deps=isort==5.9.1
-changedir={toxinidir}/examples
-commands=isort --thirdparty connexion --check-only --diff .
-
-[testenv:isort-check-tests]
-basepython=python3
-deps=isort==5.9.1
-changedir={toxinidir}/tests
-commands=isort --thirdparty aiohttp,connexion --check-only --diff .
-
-[testenv:mypy]
 deps=
-    mypy==0.910
-    types-PyYAML
-    types-requests
-    types-setuptools
-    types-ujson
-ignore_outcome=true
-commands=mypy --exclude='examples/' connexion tests
+    poetry
+allowlist_externals=
+    cp
+    sed
+    mv
+commands=
+    min: cp pyproject.toml .pyproject.toml
+    min: sed -i -E 's/"(\^|~|>=)([ 0-9])/"==\2/' pyproject.toml
```

Comment:
Is the `testenv` still used? I ran tox and did not see these commands appear.  I was checking because this `sed`  line does not work on MacOS, please see PR #2047 for a fix. But, maybe these lines are obsolete? Sorry I'm not familiar enough with how poetry works.

Reply:
Indeed, there should ideally also be a `py39-min`.

The `min` is to run the tests against the minimum required versions of the dependencies, instead of the default of taking the latest allowed version. Then, we can adjust minimum versions accordingly.

Reply:
I'm trying to find the min line you mention, am I blind ? I don't see any `pyxx-min` lines in the `tox.ini` file, especially not in `envlist`, that will trigger a test against the minimum dependencies.